src/main.rs

```diff
@@ -0,0 +1,272 @@
+    )?)
+}
+
+async fn login_token(auth: BasicAuth) -> impl Responder {
+    debug!("login_token() called");
+    let Ok(user) = std::env::var("LOGIN_USER") else {
+        error!("login_token: missing user");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+    let Ok(password) = std::env::var("LOGIN_PASSWORD") else {
+        error!("login_token: missing password");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+    if auth.user_id() != user || auth.password() != Some(password.as_str()) {
+        error!("login_token: wrong credentials");
+        return HttpResponse::build(StatusCode::UNAUTHORIZED).finish();
+    }
+    let Ok(key) = std::env::var("CENTRIFUGO_TOKEN_HMAC_SECRET_KEY") else {
+        error!("login_token: missing secret key");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+    let key = HS256Key::from_bytes(key.as_bytes());
+    let claims =
+        Claims::create(Duration::from_hours(TOKEN_EXPIRE_HOURES)).with_subject("omnect-ui");
+
+    let Ok(token) = key.authenticate(claims) else {
+        error!("login_token: cannot create token");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+
+    HttpResponse::Ok().body(token)
+}
+
+async fn refresh_token(auth: BearerAuth) -> impl Responder {
+    debug!("refresh_token() called");
+
+    let (status_code, error_msg) = verify(auth);
+
+    if status_code != StatusCode::OK {
+        error!("refresh_token verify: {error_msg}");
+        return HttpResponse::build(status_code).finish();
+    }
+
+    let Ok(key) = std::env::var("CENTRIFUGO_TOKEN_HMAC_SECRET_KEY") else {
+        error!("refresh_token: missing secret key");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+    let key = HS256Key::from_bytes(key.as_bytes());
+    let claims =
+        Claims::create(Duration::from_hours(TOKEN_EXPIRE_HOURES)).with_subject("omnect-ui");
+
+    let Ok(token) = key.authenticate(claims) else {
+        error!("refresh_token: cannot create token");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+
+    HttpResponse::Ok().body(token)
+}
+
+async fn reboot(auth: BearerAuth) -> impl Responder {
+    debug!("reboot() called");
+
+    post("/reboot/v1", Some(auth)).await
+}
+
+async fn reload_network(auth: BearerAuth) -> impl Responder {
+    debug!("reload_network() called");
+
+    post("/reload-network/v1", Some(auth)).await
+}
+
+async fn post(path: &str, auth: Option<BearerAuth>) -> HttpResponse {
+    if let Some(auth) = auth {
+        let (status_code, error_msg) = verify(auth);
+
+        if status_code != StatusCode::OK {
+            error!("put {path} verify: {error_msg}");
+            return HttpResponse::build(status_code).finish();
+        }
+    }
```

Comment:
On the long term: would it be sensible to have an actix middleware for the bearer auth?

Reply:
maybe (-;